TESTS/usb_device/basic/USBEndpointTester.cpp

```diff
@@ -0,0 +1,817 @@
+/*
+ * Copyright (c) 2018-2018, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "stdint.h"
+#include "USBEndpointTester.h"
+#include "mbed_shared_queues.h"
+#include "EndpointResolver.h"
+
+#define DEFAULT_CONFIGURATION       (1)
+
+#define NUM_PACKETS_UNTIL_ABORT     2
+#define NUM_PACKETS_AFTER_ABORT     8
+
+#define VENDOR_TEST_CTRL_IN         1
+#define VENDOR_TEST_CTRL_OUT        2
+#define VENDOR_TEST_CTRL_IN_SIZES   9
+#define VENDOR_TEST_CTRL_OUT_SIZES  10
+#define VENDOR_TEST_READ_START      11
+
+#define EVENT_READY (1 << 0)
+
+#define TEST_SIZE_EP_BULK_MAX   (64)
+#define TEST_SIZE_EP_BULK_MIN   (8)
+#define TEST_SIZE_EP_BULK_0     (16)
+#define TEST_SIZE_EP_BULK_1     TEST_SIZE_EP_BULK_MAX
+#define TEST_SIZE_EP_BULK_2     (32)
+#define TEST_SIZE_EP_BULK_3     (16)
+#define TEST_SIZE_EP_BULK_4     TEST_SIZE_EP_BULK_MIN
+
+#define TEST_SIZE_EP_INT_MAX    (64)
+#define TEST_SIZE_EP_INT_MIN    (1)
+#define TEST_SIZE_EP_INT_0      (16)
+#define TEST_SIZE_EP_INT_1      TEST_SIZE_EP_INT_MAX
+#define TEST_SIZE_EP_INT_2      (32)
+#define TEST_SIZE_EP_INT_3      (16)
+#define TEST_SIZE_EP_INT_4      TEST_SIZE_EP_INT_MIN
+
+#define TEST_SIZE_EP_ISO_MAX    (1023)
+#define TEST_SIZE_EP_ISO_MIN    (1)
+#define TEST_SIZE_EP_ISO_0      (0)
+#define TEST_SIZE_EP_ISO_1      (0)
+#define TEST_SIZE_EP_ISO_2      (0)
+#define TEST_SIZE_EP_ISO_3      (0)
+#define TEST_SIZE_EP_ISO_4      (0)
+
+#define EP_BULK_OUT 0
+#define EP_BULK_IN  1
+#define EP_INT_OUT  2
+#define EP_INT_IN   3
+#define EP_ISO_OUT  4
+#define EP_ISO_IN   5
+
+USBEndpointTester::ep_config_t USBEndpointTester::_intf_config_max[NUM_ENDPOINTS] = {
+    { false, USB_EP_TYPE_BULK, TEST_SIZE_EP_BULK_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_bulk_out) },
+    { true, USB_EP_TYPE_BULK, TEST_SIZE_EP_BULK_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_bulk_in) },
+    { false, USB_EP_TYPE_INT, TEST_SIZE_EP_INT_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_int_out) },
+    { true, USB_EP_TYPE_INT, TEST_SIZE_EP_INT_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_int_in) },
+    { false, USB_EP_TYPE_ISO, TEST_SIZE_EP_ISO_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_iso_out) },
+    { true, USB_EP_TYPE_ISO, TEST_SIZE_EP_ISO_MAX, static_cast<ep_cb_t>(&USBEndpointTester::_cb_iso_in) },
```

Comment:
Changing the size to 256 bytes here allows the `NUCLEO_F207ZG` to run tests. Since ISO isn't being tested here anyway it might be a good quick fix.

Reply:
Done.